experiment/simulation/js/power.js

```diff
@@ -0,0 +1,334 @@
+
+    if (stackDepth > maxStackDepth + 1) {
+        resultElement.style.color = 'red';
+        resultElement.textContent = 'Stack Overflow';
+
+        stackElement.style.color = 'red';
+        stackElement.innerHTML = stackHTML;
+
+        return 'Stack Overflow';
+    }
+    resultElement.style.color = 'black';
+
+    if (!step()) {
+        return stackHTML;
+    }
+
+    if (eval(`n ${basecaseOperator} ${basecase}`)) {
+        stackHTML = stackHTML.replace(newHTML, '');
+        return basecaseReturn;
+    } else {
+        const subsol = step_power(eval(recursecase), stackDepth + 1);
+
+        if (subsol === 'Stack Overflow') {
+            return 'Stack Overflow';
+        }
+        if (subsol.toString().startsWith('<')) {
+            return subsol;
+        }
+
+        resultContent = `n: ${eval(n)} | subsol: ${subsol} | sol: undefined`;
+        resultElement.textContent = resultContent;
+
+        if (!step()) {
+            return stackHTML;
+        }
+
+        const sol = eval(`${eval(variable)} ${operator} ${subsol}`);
+
+        resultContent = `n: ${eval(n)} | subsol: ${subsol} | sol: ${sol}`;
+        resultElement.textContent = resultContent;
+
+        if (!step()) {
+            return stackHTML;
+        }
+
+        stackHTML = stackHTML.substring(newHTML.length);
+        stackElement.innerHTML = stackHTML;
+
+        return sol;
+    }
+}
+
+function check_result(result) {
+    if (result === power(variable, maxStackDepth)) {
+        resultElement.style.color = 'green';
+        resultElement.textContent = "Correct Answer: " + result.toString();
+    }
+    else {
+        resultElement.style.color = 'red';
+        resultElement.textContent = 'Wrong Answer: ' + result.toString();
+    }
+
+}
+
+resetButton.addEventListener('click', reset);
+
+document.getElementById('lang-selector').addEventListener('change', function () {
+    var selectedLanguage = document.getElementById('lang-selector').value.toLowerCase();
+
+    languages.forEach(function (language) {
+        var container = document.getElementById('container-' + language);
+        container.style.display = 'none';
+    });
+
+    if (languages.includes(selectedLanguage)) {
+        var selectedContainer = document.getElementById('container-' + selectedLanguage);
+        selectedContainer.style.display = 'block';
+    } else {
+        console.log('Unsupported language: ' + selectedLanguage);
+        resetButton.disabled = true;
+        return;
+    }
+
+    resetButton.disabled = false;
+    reset();
+});
+
+runButton.addEventListener('click', function () {
+    stackElement.innerHTML = '';
+    stackHTML = '';
+    stepstoexecute = -1;
+
+    let result = step_power(maxStackDepth, 1);
+    check_result(result);
+
+    runButton.disabled = true;
+    nextButton.disabled = true;
+    prevButton.disabled = true;
+
+    console.log(basecaseOperator, basecase, basecaseReturn, recursecase, variable, operator, maxStackDepth);
```

Comment:
Once development is complete, change all the developer logs to console.debug. Only user facing logs should be printed with console.log